src/launch/service/common.py

```diff
@@ -174,3 +187,19 @@ def input_data_validation(input_data: dict) -> dict:
         input_data["skeleton"]["tag"] = SKELETON_BRANCH
 
     return input_data
+
+
+def merge_key_into_dict(source: dict, target: dict, merge_key: str, path=None):
```

Comment:
It's not clear from the definition that Path is supposed to be an optional list (of strings?). There are two ways to annotate this, pick whichever you prefer:

```suggestion
def merge_key_into_dict(source: dict, target: dict, merge_key: str, path: list[str] | None = None):
```

or 

```suggestion
from typing import Optional

def merge_key_into_dict(source: dict, target: dict, merge_key: str, path: Optional[list[str]] = None):
```

Is this supposed to return something?

A docstring (perhaps with a real example) is also probably a good idea, it's not immediately clear what the expected outcome is.